src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -411,12 +408,14 @@ public ProgrammingExerciseStudentParticipation resumeExercise(Exercise exercise,
 
     private ProgrammingExerciseStudentParticipation copyRepository(ProgrammingExerciseStudentParticipation participation) {
         if (!participation.getInitializationState().hasCompletedState(InitializationState.REPO_COPIED)) {
-            URL repositoryUrl = versionControlService.get().copyRepository(participation.getProgrammingExercise().getTemplateRepositoryUrlAsUrl(),
-                    participation.getStudent().getLogin());
-            if (Optional.ofNullable(repositoryUrl).isPresent()) {
-                participation.setRepositoryUrl(repositoryUrl.toString());
-                participation.setInitializationState(InitializationState.REPO_COPIED);
-            }
+            final var exercise = participation.getProgrammingExercise();
+            final var projectKey = exercise.getProjectKey();
+            final var username = participation.getStudent().getLogin();
+            final var repoName = RepositoryType.TEMPLATE.getName();
+            final var newRepoUrl = versionControlService.get().copyRepository(projectKey, repoName, projectKey, username).withUser(username);
```

Comment:
Exceptions are not catched, so the participation won't be updated. But in general, I agree with you, it would be great to rework the general exception handling in Artemis, this would be part of the task there (i.e. writing good exception handlers for our REST endpoints, thereby returning appropriate entities for different exceptions). 
In this case, there are only unchecked exceptions thrown by the copy method, so we shouldn't catch them, but as just described write an exception handler for them in a future PR.